### PR TITLE
Index sources

### PR DESCRIPTION
**Pull request by bbfsdev**

### CODE REVIEW

File: es/common.go [hunk 1]
```diff
@@ -29,7 +31,7 @@ var (
 
 func InitVars() {
 	pythonPath = viper.GetString("elasticsearch.python-path")
-	operatingSystem = viper.GetString("elasticsearch.os")
+	operatingSystem = viper.GetString("mdb.os")
```
Comment (edoshor) [inline es/common.go]: We don't really need a config param of our own.
We can let the [runtime pkg](https://socketloop.com/tutorials/golang-detect-os-operating-system) do that for us.

File: es/parse_docs.py [hunk 2]
```diff
@@ -5,8 +5,6 @@
 
 with open(sys.argv[1], 'rb') as f:
     document = Document(f)
-    for p in document.paragraphs:
-        sys.stdout.write(p.text.encode('utf-8'))
-        sys.stdout.flush()
-    # text = '\n'.join([p.text for p in document.paragraphs])
-    # print(text.encode('utf-8'))
+    text = '\n'.join([p.text.encode('utf-8') for p in document.paragraphs])
```
Comment (edoshor) [inline es/parse_docs.py]: This is inefficient as it loads everything into memory and concatenate strings in a loop for no apparent reason.

If you need newlines simply `sys.stdout.write('\n')`

File: es/sources_index.go [hunk 3]
```diff
@@ -0,0 +1,343 @@
...
+				if ai18n.FullName.Valid && ai18n.FullName.String != "" {
+					source.Authors = append(source.Authors, ai18n.FullName.String)
+				}
+			}
+
+			i18nMap[i18n.Language] = source
+		}
+	}
+
+	if !hasDocxForSomeLanguage {
```
Comment (edoshor) [inline es/sources_index.go]: This should not stop the index process.
Some sources in the MDB don't have any docx at all.

File: es/common.go [hunk 4]
```diff
@@ -46,6 +48,8 @@ func InitVars() {
 	}
 	docFolder = viper.GetString("elasticsearch.docx-folder")
 	utils.Must(os.MkdirAll(docFolder, 0777))
+	sourcesFolder = path.Join(viper.GetString("elasticsearch.sources-folder"))
```
[the remark this answers is not in the document]
Comment (bbfsdev) [reply es/common.go]: Done

File: es/common.go (hunk 1 above)
Comment (bbfsdev) [reply es/common.go]: Done.

File: es/sources_index.go (hunk 3 above)
Comment (bbfsdev) [reply es/sources_index.go]: Done.

**Pull request merged**